### DISCUSSION TITLE
如何自己新增一个json文件来创建新的课程呢？

### DISCUSSION THREAD

**Discussion by kkpapa** [cuixueshe/earthworm#277] [General]
目前默认是45课，通过clone代码在本地运行，想增加一些新的课程，比如命名以下文件为courses/46.json, 并且在loadCourses.json和courses.json增加了相应的第四十六课：

loadCourses.json最后增加：
 {
                "title": "第四十六课",
                "fileName": "46"
 }

course.json最后增加： {"title":"第四十六课","fileName":"46","cId":50}，

重新初始化数据库，前后端启动正常，启动后可以查看到新增加的第四十六课：
<img width="745" alt="image" src="https://github.com/cuixueshe/earthworm/assets/9346703/15714f75-659a-4656-a8bd-931d33fa0e60">
，但点击进去就是白屏没有任何反应：
<img width="1244" alt="image" src="https://github.com/cuixueshe/earthworm/assets/9346703/19e54086-8c48-40de-a16e-8f4ab8121d7b">

后端日志也没有明显报错：
[Nest] 36694  - 03/15/2024, 3:50:23 PM VERBOSE [DB] Query: select `id`, `title` from `courses` where `courses`.`id` = ? -- params: ["50"]
[Nest] 36694  - 03/15/2024, 3:50:23 PM VERBOSE [DB] Query: select `id`, `chinese`, `english`, `soundmark` from `statements` where `statements`.`course_id` = ? order by `statements`.`order` asc -- params: ["50"]

请教这种问题如何qeun解决?

**Comment by fengstats** [comment]
@kkpapa 抱歉~ 你的操作并没有太大的问题，出现这种异常是因为手动新建的课程并没有对应的 **星荣视频跳转链接** 所导致的，如图：

![1710492317470](https://github.com/cuixueshe/earthworm/assets/48991003/e0db9792-19a8-4a97-b89c-0d48c4aadb5e)

要解决也很简单，给列表获取 link 加个边界处理，比如 `courseLinks[id]?.link`，不过这个问题目前已经修复并提交到最新代码上了，你可以重新拉下代码再试试，还有问题可以继续反馈哦~ 😊

**Comment by kkpapa** [comment]
太高效了，我pull了最新的代码，现在可以正常使用了。非常感谢。

**Reply by fengstats** [reply]
哈哈能解决你的问题就好~
